File: src/Blog/Blog.Blazor/Data/Entities/SiteSettings.cs
namespace Blog.Blazor.Data.Entities;

public sealed class SiteSettings
{
    public int Id { get; set; }
    public string? ShortName { get; set; }
    public string? PortfolioUrl { get; set; }
    public bool MaintenanceEnabled { get; set; }
    public string? ContactEmail { get; set; }
    public bool ContactAvailable { get; set; }
    public string? SourceRepositoryUrl { get; set; }
    public DateTime? CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }

    public List<SiteSettingsTranslation> Translations { get; set; } = [];
    public List<ContactProfile> ContactProfiles { get; set; } = [];
}

File: laravel/app/Models/SiteSettings.php
<?php

namespace App\Models;

use App\Models\Concerns\HasTranslations;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;

/**
 * @method SiteSettingsTranslation|null translation(?string $locale = null)
 */
class SiteSettings extends Model
{
    use HasFactory, HasTranslations;

    protected $table = 'site_settings';

    protected $fillable = ['short_name', 'portfolio_url', 'maintenance_enabled', 'contact_email', 'contact_available', 'source_repository_url'];

    protected $casts = ['maintenance_enabled' => 'boolean', 'contact_available' => 'boolean'];

    /**
     * @return HasMany<SiteSettingsTranslation, $this>
     */
    public function translations(): HasMany
    {
        return $this->hasMany(SiteSettingsTranslation::class);
    }

    /**
     * @return HasMany<ContactProfile, $this>
     */
    public function contactProfiles(): HasMany
    {
        return $this->hasMany(ContactProfile::class);
    }
}
